plugins/basic/nexus-timeline-plugin/src/main/java/org/sonatype/nexus/timeline/feeds/subscribers/RepositorySubscriber.java

```diff
@@ -33,136 +41,47 @@
-  //  inspect(e);
-  //}
-  //
-  //@Subscribe
-  //@AllowConcurrentEvents
-  //public void on(final RepositoryConfigurationUpdatedEvent e) {
-  //  inspect(e);
-  //}
-  //
-  //private void inspect(Object evt) {
-  //  if (!isNexusStarted()) {
-  //    return;
-  //  }
-  //  final Repository repository;
-  //  final Date date;
-  //  if (evt instanceof RepositoryRegistryRepositoryEvent) {
-  //    RepositoryRegistryRepositoryEvent target = (RepositoryRegistryRepositoryEvent)evt;
-  //    repository = target.getRepository();
-  //    date = target.getEventDate();
-  //  }
-  //  else {
-  //    RepositoryConfigurationUpdatedEvent target = (RepositoryConfigurationUpdatedEvent)evt;
-  //    repository = target.getRepository();
-  //    date = target.getEventDate();
-  //  }
-  //
-  //  final Map<String, String> data = Maps.newHashMap();
-  //  final String action;
-  //  if (evt instanceof RepositoryRegistryEventAdd) {
-  //    action = FeedRecorder.REPO_CREATED;
-  //  }
-  //  else if (evt instanceof RepositoryRegistryEventRemove) {
-  //    action = FeedRecorder.REPO_DROPPED;
-  //  }
-  //  else {
-  //    action = FeedRecorder.REPO_UPDATED;
-  //    final RepositoryConfigurationUpdatedEvent configured = (RepositoryConfigurationUpdatedEvent) evt;
-  //    putIfNotNull(data, "localUrlChanged", String.valueOf(configured.isLocalUrlChanged()));
-  //    putIfNotNull(data, "remoteUrlChanged", String.valueOf(configured.isRemoteUrlChanged()));
-  //    putIfNotNull(data, "downloadRemoteIndexEnabled", String.valueOf(configured.isDownloadRemoteIndexEnabled()));
-  //    putIfNotNull(data, "madeSearchable", String.valueOf(configured.isMadeSearchable()));
-  //    putIfNotNull(data, "localStatusChanged", String.valueOf(configured.isLocalStatusChanged()));
-  //  }
-  //
-  //  putIfNotNull(data, "repoId", repository.getId());
-  //  putIfNotNull(data, "repoName", repository.getName());
-  //  // TODO: who changed it?
-  //  //putIfNotNull(data, "user.id", userId);
-  //  //putIfNotNull(data, "user.ip", userIp);
-  //  //putIfNotNull(data, "user.ua", userAgent);
-  //  final FeedEvent fe = new FeedEvent(
-  //      FeedRecorder.FAMILY_REPO,
-  //      action,
-  //      date,
-  //      null, // TODO: who changed it?
-  //      "/", // link to UI
-  //      data
-  //  );
-  //  getFeedRecorder().addEvent(fe);
-  //}
 
-  private boolean isNexusStarted() {
-    return systemStatusProvider.get().isNexusStarted();
+  private void onRepositoryEvent(final RepositoryEvent event, final String subType) {
+    final Map<String, String> data = Maps.newHashMap();
+    putIfNotNull(data, "repoName", event.getRepository().getName());
+    putIfNotNull(data, "repoType", event.getRepository().getType().getValue());
+    putIfNotNull(data, "repoFormat", event.getRepository().getFormat().getValue());
+    final FeedEvent feedEvent = new FeedEvent(
+        FeedRecorder.FAMILY_REPO,
+        subType,
+        new Date(),
+        "TODO", // TODO
```

Comment:
WAY too many TODO's in this PR, and leaking it into the client-facing data is particularly bad...